packages/core/addon/components/navi-visualizations/line-chart.js

```diff
@@ -259,6 +265,84 @@ export default Component.extend({
     return owner.factoryFor(registryEntry);
   }),
 
+  /**
+   * @property {Object} xGridLines - configuration of grids on x (time) axis
+   */
+  xGridLines: computed('model.firstObject', 'seriesConfig', function() {
+    const requestGrain =
+      get(this, 'model.firstObject.request.logicalTable.timeGrain.name') ||
+      get(this, 'model.firstObject.request.logicalTable.timeGrain');
+
+    const requestInterval = get(this, 'model.firstObject.request.intervals.0');
+    if (!requestInterval) {
+      return {};
+    }
+    let interval = Interval.parseFromStrings(requestInterval.start, requestInterval.end);
+
+    const getXGridLinesFormat = interval => {
+      const duration = moment.duration(interval.diffForTimePeriod('day'), 'days');
+      if (duration.asYears() > 4) {
+        return { granularity: 'year', format: 'YY' };
+      } else if (duration.asYears() > 2) {
+        return { granularity: 'quarter', format: '[Q]Q YY' };
+      } else if (duration.asMonths() > 5) {
+        return { granularity: 'month', format: 'MMM' };
+      } else if (duration.asWeeks() > 3) {
+        return { granularity: 'isoWeek', format: 'MMM D' };
+      } else {
+        return { granularity: 'day', format: '[]' };
+      }
+    };
+
+    const getXGridLines = (interval, { format, granularity } = getXGridLinesFormat(interval)) => {
+      let { start, end } = interval.asMoments();
+      if (start === undefined || end === undefined) {
+        return [];
+      }
+
+      let gridStart = start.clone();
+      if (
+        gridStart
+          .clone()
```

Comment:
Do we need to clone `gridStart` immediately after it's created by cloning `start`?